### DISCUSSION TITLE
Permission Denied to make Directory

### DISCUSSION THREAD

**Discussion by AdlyAlimin** [serversideup/spin#97] [Q&A]
hello, im new to docker and i have found this from browsing. Im using Pop OS and want to use Spin. Installation was successfully but when i want to add spin to existing project or create new spin it show permission denied.

panda@pop-os:~/Documents/Laravel$ spin new laravel mythirdapp
🔄 Downloading template...
Cloning from git@github.com:serversideup/spin-template-laravel-basic.git
Cloning into '/tmp/tmp.nuhMfkGOQK'...
remote: Enumerating objects: 417, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 417 (delta 11), reused 21 (delta 6), pack-reused 385 (from 1)
Receiving objects: 100% (417/417), 125.41 KiB | 439.00 KiB/s, done.
Resolving deltas: 100% (237/237), done.
8.3-cli: Pulling from serversideup/php
Digest: sha256:2b5b84509fa91622fcd9439a1b1c535feeec0ffed1f61c40b623b5a7a930fa58
Status: Image is up to date for serversideup/php:8.3-cli
docker.io/serversideup/php:8.3-cli

What's next:
    View a summary of image vulnerabilities and recommendations → docker scout quickview serversideup/php:8.3-cli
Creating a "laravel/laravel" project at "./mythirdapp"
Installing laravel/laravel (v11.1.4)

In ProjectInstaller.php line 69:
                              
  mkdir(): Permission denied  
                              

create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package> [<directory> [<version>]]]

any solution for this?

**Comment by jaydrogers** [comment]
I see you're using the `panda` user to run the command. Be sure the `~/Documents/Laravel` folder has write permissions by `panda`.

I just ran this inside of my WSL2 machine (Ubuntu) and I was able to get it to work.

If that doesn't work, run a `spin prune` (to clear Spin and Docker cache) and try again.

Here are my results. You can see the `jay` user is able to create the application without issue.

![Screenshot 2024-08-22 at 09 06 02@2x](https://github.com/user-attachments/assets/e2db5e22-059d-415d-aa8e-63084b30bd69)

**Reply by AdlyAlimin** [reply]
still same. permission for folder is panda

![image](https://github.com/user-attachments/assets/84b974d0-3ec3-4e95-b06c-2eebd5a8633a)

**Reply by jaydrogers** [reply]
That's very strange. Unfortunately I cannot replicate this on multiple machines.

If you want to try tinkering, go into your `~/.spin` directory. That should look identical to the Spin repository.

In there, look the `bin/spin` file and add `set -x` below the `set -e` (https://github.com/serversideup/spin/blob/79de2e0920ed2e1c4e091162c9d488e856a57f4a/bin/spin#L2). This will put the whole script into debug mode.

Run it again and it will give you more information of what's going on.

I would compare user IDs (since we're passing that to composer) to create your files.

Did it work for `myfirstapp` `mysecondapp` and fail on the third?
